### PR TITLE
feat: add build-docker target

### PR DESCRIPTION
**Pull request by elhmn**
#### This pull request adds the build-docker target

**Why ?**
We would like to build api docker images

**How ?**

- [x] adds the `build-docker` target to the Makefile
- [x] add a `Dockerfile` to build the image
- [x] add a `.docker-env`   

**Steps to verify:**
Run `make build-docker` and see if it builds your docker image
To run the image run `docker run -p 9999:9999 --env-file .docker-env yotas-api`

### CODE REVIEW

File: backend/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
+FROM golang:alpine AS builder
+
```
Comment (Sanix-Darker) [inline backend/Dockerfile:2]: Can you add some comments here please ?
in the future this file can became bigger

**Review by Sanix-Darker — APPROVED** [review]
Thanks for this PR, i left some comments !

File: backend/Dockerfile (hunk 1 above)
Comment (elhmn) [reply backend/Dockerfile:2]: Sure let me do that 👍

**Comment by elhmn** [comment]
Not sure that the comments I added are very useful but I did add them 😄 , @Sanix-Darker  can you have another look at this ?

**Comment by Sanix-Darker** [comment]
Looks ok for me @elhmn !
IF there is no more task here let's merge it ! 😎

**Pull request merged**